cypress/cypress/component/PanelSidebar/PanelSidebar.cy.tsx

```diff
@@ -0,0 +1,145 @@
+
+const getSidebarItems = (active?: boolean, disabled?: boolean): TSideBarMenuItem[] => [
+    {
+        id: "home",
+        title: "Home",
+        icon: faBars,
+        disabled,
+        items: [
+            {
+                title: "Home",
+                id: "home",
+                active,
+            },
+        ],
+    },
+    {
+        id: "settings",
+        title: "Settings",
+        icon: faCogs,
+        disabled,
+        items: [
+            {
+                title: "Settings",
+                id: "settings",
+            },
+            {
+                title: "Dropdown",
+                id: "dropdownTest",
+                children: [
+                    {
+                        title: "Dropdown test 1",
+                        id: "dropdown-test1",
+                    },
+                    {
+                        title: "Dropdown test 2",
+                        id: "dropdown-test2",
+                        active,
+                    }
+                ]
+            }
+        ],
+    },
+];
+
+describe("PanelSidebar.cy.tsx", () => {
+
+    it("icon and titles rendered correctly", () => {
+        const sidebarItems = getSidebarItems();
+        cy.mount(PanelSidebar(sidebarItems));
+
+        // Check if icon are rendered
+        cy.get('[data-icon="bars"]').should("be.visible");
+        cy.get('[data-icon="gears"]').should("be.visible");
+
+        // Check if is titles are rendered
+        cy.get("#home").invoke("text").should("equal", "Home");
+        cy.get("button[title=Settings]").click();
+        cy.get("#settings").invoke("text").should("equal", "Settings");
+    });
+
+    it("flat menu entry", () => {
+        const sidebarItems = getSidebarItems();
+        cy.mount(PanelSidebar(sidebarItems));
+
+        // Check page content changes
+        cy.get("#home").click();
+        cy.get("#pageContent").invoke("text").should("equal", "home");
+        cy.get("button[title=Settings]").click();
+        cy.get("#settings").click();
+        cy.get("#pageContent").invoke("text").should("equal", "settings");
+    });
+
+    it("nested menu entries", () => {
+        const sidebarItems = getSidebarItems();
+        cy.mount(PanelSidebar(sidebarItems));
+
+        // Check page content changes
+        cy.get("button[title=Settings]").click();
+        cy.get("#settings").click();
+        cy.get(".dropdown-toggle").click();
+        cy.get("#dropdown-test1").click();
+        cy.get("#pageContent").invoke("text").should("equal", "dropdown-test1");
+        cy.get("#dropdown-test2").click();
+        cy.get("#pageContent").invoke("text").should("equal", "dropdown-test2");
+    });
+
+    it("disabled entries", () => {
+        const sidebarItems = getSidebarItems(false, true);
+        cy.mount(PanelSidebar(sidebarItems));
+
+        // Check are disabled and page content not doesn't change
+        cy.get("button[title=Home]").should("have.attr", "disabled");
```

Comment:
i like this test, im not sure if it would be possible to at least do a click before checking the page content? because currently it doesnt try to change the page...

Reply:
If a button is disabled, the `click()` is not performed and returns an error

![image](https://github.com/neolution-ch/react-pattern-ui/assets/116279805/23e2ac2e-41b5-4e5d-9ded-4c41404ad460)

Reply:
i would say for this behavior its okay to use the {force: true} as the user technically can also still click on it 🤔

Reply:
@neotob { force: true } added

Reply:
nice many thanks!